grails-app/controllers/org/pih/warehouse/user/UserController.groovy

```diff
@@ -272,6 +273,29 @@ class UserController {
         }
     }
 
+    def changePassword() {
+        User user = userGormService.get(params?.id)
+        if (user) {
+            try {
+                userService.changePassword(user, params?.password, params?.passwordConfirm)
+                flash.message = "${warehouse.message(code: 'default.updated.message', args: [warehouse.message(code: 'user.label'), user.id])}"
+                redirect(action: "edit", id: user.id)
+            } catch (ValidationException e) {
+                // This read function is used to avoid getting lazy initialization exceptions in
+                // rendering the edit page, it is done like in the update function above
+                user = User.read(params.id)
```

Comment:
### Good practices:

If we considered good practices, it shouldn't be the way to go (solving LazyInitializationException by re-fetching an entity), but join the needed associations at the query level.
Seeing this code, something tells me that we must have some kind of OSIV mechanism that is associated with a session, and maybe it is turned off when catching an exception? :thinking: 
I've mentioned it recently in my PR: https://github.com/openboxes/openboxes/pull/4435#discussion_r1438102346
So in a nutshell - I would expect that when we have the OSIV filter disabled, we should get the Lazy every time we want to access some associations when not in a transaction boundary. 
I'd consider creating a spike ticket to investigate this part, because it's weird that in the `catch` we lose the associations.
Or maybe it's maybe related to `render` vs `redirect`? 
Could you check that?

### Not considering good practices:
Obviously we won't experience any performance issues due to that additional fetch, so I won't force you to go with the join query route, especially if you had just followed the code from the method above.